### PR TITLE
Port `ArrayRemove`, `ArrayRemoveN`, `ArrayRemoveAll` to `functions-array` subcrate

### LINKED ISSUE #9635: Port `ArrayRemove`, `ArrayRemoveN`, `ArrayRemoveAll` to `functions-array` subcrate
### Is your feature request related to a problem or challenge?

`ArrayRemove`, `ArrayRemoveN`, `ArrayRemoveAll` functions need to be ported to new sfunctions-arrays subcreate in terms of https://github.com/apache/arrow-datafusion/issues/9285.

### Describe the solution you'd like

Port `ArrayRemove`, `ArrayRemoveN`, `ArrayRemoveAll` to `functions-array` subcrate

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by erenavsarogullari**
## Which issue does this PR close?
Closes #9635.

## What changes are included in this PR?
This PR aims to do following changes in terms of Epic https://github.com/apache/arrow-datafusion/issues/9285:
`ArrayRemove`, `ArrayRemoveN`, `ArrayRemoveAll` are ported to `functions-array` subcrate.

## Are these changes tested?
Yes, all `array.slt` and `unnest.slt` based `ArrayRemove`, `ArrayRemoveN`, `ArrayRemoveAll` tests are passed.

## Are there any user-facing changes?
No

### CODE REVIEW

**Comment by erenavsarogullari** [comment]
Current `Rust / cargo test (win64)` build stage failure seems to be related environmental. We can re-trigger.
```
D:\a\_temp\45213694-13e6-4eb8-97ae-555da09c8695.sh: line 2: /c/Users/runneradmin/.cargo/bin/rustup: Device or resource busy
Error: Process completed with exit code 126.
```

File: datafusion/proto/tests/cases/roundtrip_logical_plan.rs [hunk 1]
```diff
@@ -591,10 +591,6 @@ async fn roundtrip_expr_api() -> Result<()> {
             lit(2),
         ),
         array_positions(make_array(vec![lit(4), lit(3), lit(3), lit(1)]), lit(3)),
-        array_except(
-            make_array(vec![lit(1), lit(2), lit(3)]),
-            make_array(vec![lit(1), lit(2)]),
-        ),
```
Comment (jayzhan211) [inline datafusion/proto/tests/cases/roundtrip_logical_plan.rs:597]: I think you accidentally remove it. Also, add remove's too

Comment (erenavsarogullari) [reply datafusion/proto/tests/cases/roundtrip_logical_plan.rs:597]: Thanks for the review, addressed.

**Comment by jayzhan211** [comment]
> Current `Rust / cargo test (win64)` build stage failure seems to be related environmental. We can re-trigger this stage if possible.
> 
> ```
> D:\a\_temp\45213694-13e6-4eb8-97ae-555da09c8695.sh: line 2: /c/Users/runneradmin/.cargo/bin/rustup: Device or resource busy
> Error: Process completed with exit code 126.
> ```

It seems like an issue from github

https://github.com/rust-lang/rustup/issues/3709

File: datafusion/functions-array/src/utils.rs [hunk 2]
```diff
@@ -108,6 +111,107 @@ pub(crate) fn align_array_dimensions<O: OffsetSizeTrait>(
...
+/// ```text
+/// compare_element_to_list(
+///     [1, 2, 3], [1, 2, 3], 0, true => [true, false, false]
+///     [1, 2, 3, 3, 2, 1], [1, 2, 3], 1, true => [false, true, false, false, true, false]
+///
+///     [[1, 2, 3], [2, 3, 4], [3, 4, 5]], [[1, 2, 3], [2, 3, 4], [3, 4, 5]], 0, true => [true, false, false]
+///     [[1, 2, 3], [2, 3, 4], [2, 3, 4]], [[1, 2, 3], [2, 3, 4], [3, 4, 5]], 1, false => [true, false, false]
+/// )
+/// ```
+pub(crate) fn compare_element_to_list(
```
Comment (jayzhan211) [inline datafusion/functions-array/src/utils.rs:141]: There are three duplicated code including this one, can you remove other two?

**Review by jayzhan211 — APPROVED** [review]
LGTM

**Comment by jayzhan211** [comment]
@erenavsarogullari Let me merge this first, those duplication can be removed later

**Comment by erenavsarogullari** [comment]
@jayzhan211 Thanks for the merge. I will address duplication by separate PR.

**Pull request merged**